src/log/log.c

```diff
@@ -1456,7 +1492,26 @@ __log_truncate(WT_SESSION_IMPL *session, WT_LSN *lsn, bool this_log)
 	for (i = 0; i < logcount; i++) {
 		WT_ERR(__wt_log_extract_lognum(session, logfiles[i], &lognum));
 		if (lognum > lsn->l.file && lognum < log->trunc_lsn.l.file) {
-			WT_ERR(__log_openfile(session, lognum, 0, &log_fh));
+			opened = false;
+			if (salvage_mode) {
+				WT_ERR(__log_open_verify(session, lognum,
+				    &log_fh, NULL, NULL, true, &need_salvage));
+				if (need_salvage) {
+					WT_ERR(__wt_close(session, &log_fh));
```

Comment:
Is this correct? In `log_open_verify` the only time `need_salvage` comes back true, we've already set `log_fh` to NULL and called `wt_close` on it.

Reply:
That's right!  Thanks, I fixed it.